qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -1547,6 +1602,32 @@ def _on_load_finished(self, ok: bool) -> None:
                 emit_before_load_started=False))
             self._reload_url = None
 
+    @pyqtSlot(QWebEnginePage.LifecycleState)
```

Comment:
@The-Compiler I got stuck trying to make this work.

What i am seeing is:
```
TypeError: connect() failed between recommendedStateChanged(QWebEnginePage::LifecycleState) and _check_recommended_lifecycle_state()
```

Which is weird given that the [signal](https://doc.qt.io/qt-5/qwebenginepage.html#recommendedState-prop) is there and matches the signature i am providing.

Any ideas?

Reply:
If you look at `_on_feature_permission_requested` for example it points to the enum with a string like `'QWebEnginePage::Feature'`, also make sure you have a new enough pyqt.

Reply:
I tried using `'QWebEnginePage::LifecycleState'` to no avail. 

>  also make sure you have a new enough pyqt.

Not sure i understand what "enough pyqt" means in this case.

Reply:
Might need to do `@pyqtSlot(object)  # QWebEnginePage.LifecycleState` here to keep compatibility with older PyQt versions which don't know about that type.

Reply:
I did that and still seeing the same error. Not sure i have a well enough understanding on why this happens. I can see there is another slot which defines:
```
    @pyqtSlot(QWebEnginePage.RenderProcessTerminationStatus, int)
```

Which works.

Reply:
Then I'm assuming (like @toofar said) your PyQt, Qt, or the Qt you compiled PyQt against is too old - see `:version` and make sure all Qt version numbers are 5.14 or newer.

Reply:
This is what i see:
```
Qt: 5.14.2

CPython: 3.8.2
PyQt: 5.14.2

PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.14.0
PyQt5.QtWebKitWidgets: no
```

Reply:
@The-Compiler Does it make a difference that this signal is defined in `QWebEnginePrivate`?
https://code.qt.io/cgit/qt/qtwebengine.git/tree/src/webenginewidgets/api/qwebenginepage.cpp?h=5.15#n1887